systemservices/marketplace/src/tasks/publishPurchase.ts

```diff
@@ -0,0 +1,24 @@
+import { TaskInputs, TaskOutputs } from "mesg-js/lib/service"
+import Web3 from "web3"
+import { decodeLog } from "../contracts/utils";
+import { eventHandlers } from "../events";
+
+export default (
+  web3: Web3,
+) => async (inputs: TaskInputs, outputs: TaskOutputs): Promise<void> => {
+  try {
+    let receipt = null
+    for (let i=0; i<inputs.signedTransactions.length; i++) {
+      receipt = await web3.eth.sendSignedTransaction(inputs.signedTransactions[i])
+    }
+    if (receipt === null || receipt.logs === undefined) throw new Error('receipt does not contain logs')
+    const eventHandler = eventHandlers['ServicePurchased']
+    const decodedLog = decodeLog(web3, eventHandler.abi, receipt.logs[receipt.logs.length - 1])
```

Comment:
I feel its a bit hacky to take the last log. Can't we just iterate over the logs and filter for the one who match the event we want ?

Reply:
Updated by filtering the array of logs.